UN-OCHA/styleguide
# CD-28: search

Html, CSS and JS for this variant

Screenshots on how it should appear on [JIRA](https://humanitarian.atlassian.net/browse/CD-28)

We recently concluded that the behaviour should mirror existing inline searches used
https://www.humanitarianresponse.info and https://humanitarian.id/landing
On desktop, the search expands to fill the available space by default
But instead of it remaining visible in the header on tablet and mobile, we agreed we would use the collapsed style of the previous search.

I use class name `search--input` on main wrapper and most children elements to distinguish from the original search, and also copied most of the original search's style to get the same mobile/tablet behaviour

### To test:

1. Checkout this PR
2. Compile styleguide
3. Visit http://localhost:4000/common-design/components.html and click the Inline Search demo link
4. Confirm the inline search works for desktop and works like the original search on tablet and mobile
5. Also note the changes to the styleguide itself (see commits) where I added some conditionals in order to create ademo page for the inline search
